src/allotropy/allotrope/schemas.py

```diff
@@ -25,3 +26,23 @@ def add_definitions(schema: dict[str, Any]) -> dict[str, Any]:
 def get_schema(schema_relative_path: str) -> dict[str, Any]:
     with open(os.path.join(SCHEMAS_DIR, schema_relative_path)) as f:
         return add_definitions(json.load(f))
+
+
+def get_schema_path_from_manifest(manifest: str) -> str:
+    match = re.match(r"http://purl.allotrope.org/manifests/(.*)\.manifest", manifest)
+    if not match:
+        msg = f"No matching schema in repo for manifest: {manifest}"
+        raise ValueError(msg)
+    return f"{match.groups()[0]}.json"
+
+
+def get_schema_from_manifest(manifest: str) -> dict[str, Any]:
+    return get_schema(get_schema_path_from_manifest(manifest))
+
+
+def get_schema_from_model(model: Any) -> dict[str, Any]:
+    manifest = getattr(model, "manifest", getattr(model, "field_asm_manifest", None))
+    if not manifest:
+        msg = f"No 'manifest' or 'field_asn_manifest' found in model: {type(model)}"
```

Comment:
```suggestion
        msg = f"No 'manifest' or 'field_asm_manifest' found in model: {type(model)}"
```

Reply:
I swear I... try to spell things correctly 😭 😭